source/features/useful-not-found-page.tsx

```diff
@@ -28,119 +25,360 @@ async function checkAnchor(anchor: HTMLAnchorElement): Promise<void> {
+
+	if (!currentBranch) {
 		return;
 	}
 
 	const defaultBranch = await getDefaultBranch();
-	if (branch === defaultBranch) {
+	if (currentBranch === defaultBranch) {
 		return;
 	}
 
-	parts[3] = defaultBranch; // Change branch
-	const url = '/' + parts.join('/');
+	// Change branch
+	parts[3] = defaultBranch;
+	// We need to view commits
+	parts[2] = "commits";
+	// Create the url
+	const url = "/" + parts.join("/");
+	// Check if that path exists
 	if (await is404(location.origin + url)) {
 		return;
 	}
 
-	bar.after(
-		<p className="container mt-4 text-center">
-			The {getType()} exists on the <a href={url}>default branch</a>.
-		</p>,
-	);
+	// If it does, return it
+	return url;
 }
 
-function init(): false | void {
-	const parts = parseCurrentURL();
-	if (parts.length <= 1 || !select.exists('[alt*="This is not the web page you are looking for"]')) {
-		return false;
+async function whatHappenedToTheFile(): Promise<
+	Record<string, string | undefined> | undefined
+> {
+	// First, check if the file exists on the default branch
+	const urlOnDefaultBranch = await getUrlToFileOnDefaultBranch();
+	if (urlOnDefaultBranch) {
+		return {
+			type: "isOnDefaultBranch",
+			linkToLastVersion: urlOnDefaultBranch,
+		};
 	}
 
-	const bar = <h2 className="container mt-4 text-center"/>;
+	// Else, check if the file exists in commit history
+	const lastCommitIncludingFile = await getLastCommitForFile();
+	if (lastCommitIncludingFile) {
+		return lastCommitIncludingFile;
+	}
+
+	// Else, check if the file exists in the default branch's history
+	const linkToCommitHistoryOnDefaultBranch =
+		await getLinkToCommitHistoryOnDefaultBranch();
+	if (linkToCommitHistoryOnDefaultBranch) {
+		return {
+			type: "wasOnDefaultBranch",
+			linkToCommitHistory: linkToCommitHistoryOnDefaultBranch,
+		};
+	}
 
+	return undefined;
+}
+
+async function showAdditionalInformation(bar: Element): Promise<void> {
+	// Check what happened
+	const eventThatHappened = await whatHappenedToTheFile();
+	if (!eventThatHappened) {
+		return;
+	}
+
+	// Parse that and do/show something
+	if (eventThatHappened.type === "isOnDefaultBranch") {
+		bar.after(
+			<p className="container mt-4 text-center">
+				<a href={eventThatHappened.linkToLastVersion}>This file</a> only exists
+				on the default branch.
+			</p>
+		);
+		return;
+	}
+
+	if (eventThatHappened.type === "wasOnDefaultBranch") {
+		bar.after(
+			<p className="container mt-4 text-center">
+				This file used to exist on the default branch - see{" "}
+				<a href={eventThatHappened.linkToCommitHistory}>commit history</a>.
+			</p>
+		);
+		return;
+	}
+
+	if (eventThatHappened.type === "removed") {
+		bar.after(
+			<p className="container mt-4 text-center">
+				<a href={eventThatHappened.linkToLastVersion}>This file</a> was deleted
+				by{" "}
+				<a href={eventThatHappened.linkToCommit}>
+					{eventThatHappened.commitSha}
+				</a>{" "}
+				(<relative-time datetime={new Date(eventThatHappened.commitDate!)} />) -
```

Comment:
Let's skip the commit number, we can just linkify the date (everywhere)
```ts
				<a href={eventThatHappened.linkToCommit}>
					<relative-time datetime={new Date(eventThatHappened.commitDate!)} />
				</a> -
```

Reply:
Sure